cosmos/lib/cosmos/api/target_api.rb

```diff
@@ -74,5 +79,73 @@ def get_all_target_info(scope: $cosmos_scope, token: $cosmos_token)
       end
       info
     end
+
+    # Read a file from a target
+    #
+    # @param [String] Path to a file in a target directory
+    # @param [String] File mode, default is 'r' but you can also pass 'rb' for binary data
+    # @return [String|nil]
+    def read_target_file(path, mode = 'r', scope: $cosmos_scope, token: $cosmos_token)
+      authorize(permission: 'system', scope: scope, token: token)
+      file = nil
+      local_path = File.join(Dir.tmpdir, 'cosmos', 'target_files', path)
+      FileUtils.mkdir_p(File.dirname(local_path))
+      client = Aws::S3::Client.new
+      begin
+        Cosmos::Logger.info "Reading #{scope}/targets_modified/#{path}, mode #{mode}"
+        client.get_object(bucket: "config", key: "#{scope}/targets_modified/#{path}", response_target: local_path)
```

Comment:
These methods should return a Tempfile object, not the data directly.   The tempfile object is more versatile and can handle large files on systems with low RAM.   Also a Ruby Tempfile object should cleanup the file automatically when garbage collected.   As written, the files will just waste space forever.

Reply:
You can't return a Ruby object over an API. The file object only exists on the server so how can the client interact with it?